File: TI1/TI1.4 RESTFUL_DOTNET_SINBDD_GR09/02.CLIMOV/CONUNI_RESTFUL_DOTNET_CLIMOV_G09/app/src/main/java/ec/edu/monster/modelo/ConversionResponse.kt
package ec.edu.monster.modelo

import com.google.gson.annotations.SerializedName

/**
 * Modelo para respuestas de conversión RESTful
 */
data class ConversionResponse(
    @SerializedName("ValorOriginal")
    val valorOriginal: Double,
    @SerializedName("ValorConvertido")
    val valorConvertido: Double,
    @SerializedName("UnidadOrigen")
    val unidadOrigen: String,
    @SerializedName("UnidadDestino")
    val unidadDestino: String,
    @SerializedName("Categoria")
    val categoria: String,
    @SerializedName("Exito")
    val exito: Boolean,
    @SerializedName("Mensaje")
    val mensaje: String? = null,  // Campo del servidor .NET (mayúscula en JSON)
    @SerializedName("Timestamp")
    val timestamp: Long = 0
)

File: TI1/TI1.2 SOAP_DOTNET_SINBDD_GR09/02.CLIESC/CONUNI_SOAP_DOTNET_CLIESC_G09/CONUNI_SOAP_DOTNET_CLIESC_G09/modelo/ConversionResponse.cs
namespace CONUNI_SOAP_DOTNET_CLIESC_G09.modelo
{
    /// <summary>
    /// Modelo para las respuestas de conversión
    /// </summary>
    public class ConversionResponse
    {
        public double ValorOriginal { get; set; }
        public double ValorConvertido { get; set; }
        public string UnidadOrigen { get; set; }
        public string UnidadDestino { get; set; }
        public string Categoria { get; set; }
        public bool Exito { get; set; }
        public string Mensaje { get; set; }
        public long Timestamp { get; set; }
    }
}


